File: leetcode/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapping: Dict[int, int] = {}
        for i, n in enumerate(nums):
            m = target - n
            if m in mapping:
                return mapping[m], i
            mapping[n] = i

File: leetcode/two-sum.go
func twoSum(nums []int, target int) []int {
    mapping := map[int]int{}
    for i, n := range nums {
        if p, ok := mapping[target-n]; ok {
            return []int{p, i}
        }
        mapping[n] = i
    }
    return nil
}
